pkg/arvo/app/ping.hoon

```diff
@@ -48,256 +34,39 @@
-        |=  =ship
-        ?:  =(our ship)
-          ~
-        =/  wire  /pub/(scot %uw nonce.state)/ping/(scot %p ship)
-        `u=[%pass wire %agent [ship %ping] %poke %noun !>(~)]
-      ::
-      ++  take-pings
-        |=  [=wire error=(unit tang)]
-        ^-  (quip card _state)
-        ?.  ?=([%pub @ %ping @ ~] wire)  `state
-        ?.  (once i.t.wire)  `state
-        =/  ship  (slav %p i.t.t.t.wire)
-        %-  (slog ?~(error ~ ['ping: got nack' >ship< u.error]))
-        `state
-      ::
-      ++  check-ip
-        ^-  (quip card _state)
-        :_  state
-        =/  wire  /pub/(scot %uw nonce.state)/ip
-        =/  =request:http  [%'GET' ip-reflector ~ ~]
-        [%pass wire %arvo %i %request request *outbound-config:iris]~
-      ::
-      ++  take-ip
-        |=  [our=@p =wire resp=client-response:iris]
-        ^-  (quip card _state)
-        ?.  ?=([%pub @ %ip ~] wire)  `state
-        ?.  (once i.t.wire)  `state
-        ::
-        ?.  ?=(%finished -.resp)  `state  :: will retry in a minute
-        ?.  ?=(%200 status-code.response-header.resp)
-          =*  s  status-code.response-header.resp
-          %-  (slog leaf+"ping: ip check failed: {<s>}" ~)
-          `state
-        ::
-        ?~  full-file.resp
-          %-  (slog 'ping: ip check body empty' ~)
-          `state
-        ::
-        =*  body  q.data.u.full-file.resp
-        ?~  body
-          %-  (slog 'ping: ip check body empty' ~)
-          `state
-        ::
-        =/  ip  (end [3 (dec (met 3 body))] body)
-        ?:  =(ip.plan.state `ip)  `state
-        ::
-        =.  ip.plan.state  `ip
-        (send-pings our)
-      ::
-      ++  set-timer
-        |=  now=@da
-        ^-  (quip card _state)
-        =/  =wire  /pub/(scot %uw nonce.state)/wait
-        [[%pass wire %arvo %b %wait (add ip-timeout now)]~ state]
-      ::
-      ++  take-wait
-        |=  [our=@p now=@da =wire]
-        ^-  (quip card _state)
-        ?.  ?=([%pub @ %wait ~] wire)  `state
-        ?.  (once i.t.wire)  `state
-        ;<  new-state=_state  rind  check-ip
-        =.  state  new-state
-        ::
-        (set-timer now)
-      --
-    ::  Subsystem for formally acknowledging a change in our IP:PORT
-    ::
-    ::    If our sponsor sends a STUN response, with an IP different than what
-    ::    we had previously cached, we formally acknowledge this change by
-    ::    sending one %poke to every ship in the sponsorship chain.
-    ::
-    ++  one
-      ?>  ?=(%one -.plan.state)
-      |%
-      ++  kick
-        |=  our=@p
-        ^-  (quip card _state)
-        :_  state
-        %-  ~(rep in ships.state)
-        |=  [=ship cards=(list card)]
-        ?:  =(our ship)  cards
-        =/  wire  /one/(scot %uw nonce.state)/ping/(scot %p ship)
-        :_  cards  ^-  card
-        [%pass wire %agent [ship %ping] %poke %noun !>(~)]
-      --
-    --
+  ++  galaxy-for
+    |=  [=ship =bowl:gall]
+    ^-  @p
+    =/  next  (sein:title [our now our]:bowl)
```

Comment:
This needs to be `(sein:title ship [now our]:bowl)` otherwise we bail:meme recursing forever and the first %stun %stop will fail.